RIOT-OS/RobotFW-tests
# tests/periph_gpio: Add periph gpio tests

## Contribution Description

 robotframework: add noncritical if skip as tag

Change the makefile python robot call to force noncritical tests when skip tag is present.


 tests/periph_gpio: test all additional periphs gpios

Add in a test that uses the board environments to determine if a pin should be connected.
If pin information is present disable the module and test its pins.
If pins are not present a skip tag will be added which will mark it as noncritical.


## Testing Procedure:

Connect the board you want to test with a PHiLIP Board according to the environment file.
Present pins are tested and should pass if wired correctly.
Pins that are not present should fail as a noncritical test and therefore the test as a whole still succeeds.

run `BOARD=<my_board> make flash robot-test -C tests/periph_gpio`
if the test passes try disconnecting a pin to see that it fails.
